etna/analysis/plotters.py

```diff
@@ -1958,3 +1959,114 @@ def update(**kwargs):
         plt.show()
 
     interact(update, **sliders)
+
+
+class ComponentsMode(str, Enum):
+    """Enum for components plotting modes."""
+
+    per_component = "per-component"
+    together = "together"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise NotImplementedError(
+            f"{value} is not a valid {cls.__name__}. Supported modes: {', '.join([repr(m.value) for m in cls])}"
+        )
+
+
+def plot_forecast_decomposition(
+    forecast_ts: "TSDataset",
+    test_ts: Optional["TSDataset"] = None,
+    mode: str = ComponentsMode.per_component,
```

Comment:
It isn't really a correct typing because `ComponentsMode.per_component` isn't a string. May be `ComponentsMode.per_component.value` would work.

Reply:
Changed